src/server/bg_services/md_aggregator.js

```diff
@@ -30,13 +30,17 @@ async function run_md_aggregator(md_store, system_store, target_now, delay) {
     }
 
     let has_more = true;
+    let update_range = true;
+    let range = {};
 
     while (has_more) {
-        const range = await find_next_range({ target_now, system_store });
-        const update = range && await range_md_aggregator({ md_store, system_store, range });
-        if (update) {
+        if (update_range) range = await find_next_range({ target_now, system_store });
+        const changes = range && await range_md_aggregator({ md_store, system_store, range });
```

Comment:
The range is declared in line 34 above.
This will always exist so expression: ```range &&``` will always be validated as true.

Reply:
This way you will call the ```range_md_aggregator``` with ```range = {}``` when ```update_range``` is false.
Inside ```range_md_aggregator``` you will get ```from_time``` and ```till_time``` which will be undefined.
Then inside the md_store you will attempt to create a time inside method ```make_md_id_from_time``` that will fail since it will attempt to work on undefined values.

Reply:
you are correct range will allways be defined (as in before my PR) - not sure why it was there in the first place... can be removed. you can't call range_md_aggregator with {}. it just suppose not to update a range that wasn't changed.

Reply:
```let update_range = true;```
I've somehow missed that. I understand that you want to reuse the old range in case you have the partial update so this makes sense. What's the case when you do not have ```range &&```?
I think this is redundant.

Reply:
redundant i agree :)